test/clj/rems/api/test_workflows.clj

```diff
@@ -96,6 +96,69 @@
           (is (= {:id id :enabled false :archived true}
                  (select-keys workflow [:id :enabled :archived]))))))))
 
+(deftest workflows-update-test
+  (let [api-key "42"
+        user-id "owner"
+        wf-spec {:organization "abc"
+                 :title "dynamic workflow"
+                 :type :dynamic
+                 :handlers ["bob" "carl"]}
+        wfid (-> (request :post "/api/workflows/create")
+                 (json-body wf-spec)
+                 (authenticate api-key user-id)
+                 handler
+                 read-ok-body
+                 :id)
+        ;; this is a subset of what we expect to get from the api
+        expected {:id wfid
+                  :organization "abc"
+                  :title "dynamic workflow"
+                  :workflow {:type "workflow/dynamic"
+                             :handlers ["bob" "carl"]}
+                  :enabled true
+                  :expired false
+                  :archived false}
+        fetch (fn []
+                (let [wfs (-> (request :get "/api/workflows" {:archived true :expired true :disabled true})
+                              (authenticate api-key user-id)
+                              handler
+                              read-ok-body)]
+                  (select-keys
+                   (first (filter #(= wfid (:id %)) wfs))
+                   (keys expected))))
+        update #(-> (request :put "/api/workflows/update")
+                    (json-body (merge {:id wfid} %))
+                    (authenticate api-key user-id)
+                    handler
+                    read-ok-body)]
+    (testing "before changes"
+      (is (= expected (fetch))))
+    (testing "disable and archive"
+      (is (:success (update {:enabled false :archived true})))
+      (is (= (assoc expected
+                    :enabled false
+                    :archived true)
+             (fetch))))
+    (testing "re-enable"
+      (is (:success (update {:enabled true})))
+      (is (= (assoc expected
+                    :archived true)
+             (fetch))))
```

Comment:
Could also add test for "unarchive". That would avoid the need for `(assoc :archived true)` from the following tests.